### PR TITLE
Add example and modify document for class Color

### PR DESCRIPTION
**Pull request by AzulRadio**
# Update Document

Add an example for class Color as one goal mentioned in #300

## Summary
Add an example for class Color and change the existing documents.

## Test it
```
cd example
mkdir build
cmake ../
make
./color_example
```

## Others

I found something weird about the `ignition::math::Color::HSV()` and `ignition::math::Color::SetFromHSV()` when writing the example. When using the return value of the `HSV()` as the input for `SetFromHSV()`, the value of the instance is inconsistent. But I am not sure if it is a bug or I'm just using it the wrong way so I am not sure if making it a separate issue is a good idea.

Like this (put it in /example)
```
#include <iostream>
#include <ignition/math/Color.hh>

int main() 
{
  ignition::math::Color a = ignition::math::Color(0.5, 0.6, 0.7);
  std::cout << "The RGBA value of a: " << a << std::endl;
  std::cout << "The HSV value of a: " << a.HSV() << std::endl;
  
  // use its own HSV value to set HSV value, should be exactly the same.
  a.ignition::math::Color::SetFromHSV(a.HSV()[0], a.HSV()[1], a.HSV()[2]);
  std::cout << "The RGBA value of a: " << a << std::endl;
  std::cout << "The HSV value of a: " << a.HSV() << std::endl;

  // one more time
  a.ignition::math::Color::SetFromHSV(a.HSV()[0], a.HSV()[1], a.HSV()[2]);
  std::cout << "The RGBA value of a: " << a << std::endl;
  std::cout << "The HSV value of a: " << a.HSV() << std::endl;
}
```

 
## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: examples/color_example.cc [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
```
Comment (ahcorde) [inline examples/color_example.cc:2]:
```suggestion
 * Copyright (C) 2021 Open Source Robotics Foundation
```

File: examples/color_example.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ *
+*/
+//! [complete]
+#include <iostream>
+#include <ignition/math/Color.hh>
+
+int main(int argc, char **argv) 
+{
+
+//! [Create a color]
```
Comment (ahcorde) [inline examples/color_example.cc:24]: Do you plan to add a Color tutorial ? These kind of tags are used there to insert code in the tutorial.

For example [triangle tutorial](https://github.com/ignitionrobotics/ign-math/blob/ign-math6/tutorials/triangle.md) and [online triangle tutorial](https://ignitionrobotics.org/api/math/6.9/angle.html)

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: examples/color_example.cc (hunk 2 above)
Comment (AzulRadio) [reply examples/color_example.cc:24]: Oh, I didn't know what they do but sounds like a good idea. Now we have the tutorial. Thanks for the suggestion.

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
Thank you for your contribution!

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**